### PR TITLE
Claim all Alpha token rewards for user as a lender and borrower then create receipt

### PR DESCRIPTION
**Pull request by mizu-mochi**
- Store Alpha multiplier instead of latest reward

### CODE REVIEW

File: contracts/AlToken.sol [hunk 1]
```diff
@@ -101,6 +101,14 @@ contract AlToken is ERC20, Ownable, IAlphaReceiver {
     return balanceOf(_account).mul(alphaRewardMultiplier).div(1e12).sub(latestRewards[_account]);
   }
 
+  /**
+   * @dev claim user's pending Alpha rewards by owner
+   * @param _account the user account address
+   */
+  function claimCurrentAlphaRewardByOwner(address _account) external onlyOwner {
+    claimCurrentAlphaReward(_account);
```
Comment (false10) [inline contracts/AlToken.sol:110]: Add `setLatestReward` after claim.

**Review by false10 — CHANGES_REQUESTED** [review]

File: contracts/AlToken.sol (hunk 1 above)
Comment (mizu-mochi) [reply contracts/AlToken.sol:110]: Add on `claimCurrentAlphaReward`

File: contracts/AlToken.sol [hunk 2]
```diff
@@ -59,7 +61,7 @@ contract AlToken is ERC20, Ownable, IAlphaReceiver {
   function mint(address _account, uint256 _amount) external onlyOwner {
     claimCurrentAlphaReward(_account);
     _mint(_account, _amount);
-    setLatestReward(_account);
+    setLatestAlphaMultiplier(_account);
```
Comment (false10) [inline contracts/AlToken.sol]: No need to setLatestAlphaMultiplier?

File: contracts/AlToken.sol [hunk 3]
```diff
@@ -71,7 +73,7 @@ contract AlToken is ERC20, Ownable, IAlphaReceiver {
   function burn(address _account, uint256 _amount) external onlyOwner {
     claimCurrentAlphaReward(_account);
     _burn(_account, _amount);
-    setLatestReward(_account);
+    setLatestAlphaMultiplier(_account);
```
Comment (false10) [inline contracts/AlToken.sol]: ditto

File: contracts/AlToken.sol [hunk 4]
```diff
@@ -120,18 +131,19 @@ contract AlToken is ERC20, Ownable, IAlphaReceiver {
       lendingPool.distributor().alphaToken().approve(address(vestingAlpha), pending);
       vestingAlpha.accumulateAlphaToUser(_account, pending);
     }
+    setLatestAlphaMultiplier(_account);
```
Comment (false10) [inline contracts/AlToken.sol]: I think this internal function not necessary anymore.

File: contracts/AlToken.sol [hunk 5]
```diff
@@ -152,7 +164,7 @@ contract AlToken is ERC20, Ownable, IAlphaReceiver {
     claimCurrentAlphaReward(_to);
     super._transfer(_from, _to, _amount);
     require(lendingPool.isAccountHealthy(_from), "Transfer tokens is not allowed");
-    setLatestReward(_from);
-    setLatestReward(_to);
+    setLatestAlphaMultiplier(_from);
+    setLatestAlphaMultiplier(_to);
```
Comment (false10) [inline contracts/AlToken.sol]: ditto

**Review by false10 — CHANGES_REQUESTED** [review]

**Review by false10 — APPROVED** [review]

**Pull request merged**